### ISSUE TITLE
Számlák szűrése a sandbox API-n

### ISSUE THREAD

**Issue by Razoralpha** [voov/Billingo-API-Connector#13]
Érdekes módon bármi alapján próbáltam előkeríteni a generált proformákat, mondjuk payment method, vagy számlatömb azonosító, egyik sem vezetett eredményre, ugyanúgy kirakott minden találatot, hiába kerestem a banki átutalás azonosítójára (2), továbbra is megjelentek a bankkártyás számlák is.

**Comment by albertborsos** [comment]
Szia @Razoralpha! Sikerült esetleg rájönnöd, hogyan lehet szűrni a találati listát? :D

**Comment by Razoralpha** [comment]
Áhh, jelenleg ha megadom a paramétereket akkor "NULL" választ kapok még productionben is, és ha még ez lenne az egyetlen hiba!
```
return $this->billingoAPIConnection->get("invoices/query", [
	"block" => $this->invoiceBlocks["wiretransfer"],
	"page" => 1,
	"max_per_page" => 50
]);
```
Simán a query stringbe írva legalább választ kapok, csak nem szűrtet:
```
$this->billingoAPIConnection->get("invoices/query?block=".$this->invoiceBlocks["wiretransfer"]."&page=1&max_per_page=50");
```
Lehet, hogy az éles site-on kéne meglesni, hogy mi az isten történik. :D

**Comment by albertborsos** [comment]
Köszi! Nem hiszem, hogy különbözik a production-től az api. Szerintem csak másik DB-t használ, ha a sandbox mode be van kapcsolva. Legalábbis én úgy csináltam volna.

Te akkor hogyan szűrsz? eredményhalmazon `array_filter`?

**Comment by Razoralpha** [comment]
Amikor én írtam és könyörögtem megoldásért akkor többször is az volt a válasz, hogy ez épp rossz sandbox környezetben, de állításuk szerint productionben jó lett volna, szóval valószínűleg különbözik a két api valamennyire.

Igen, az ID és esélyes, hogy a DB is eltér mögöttük, de sajnos ha épp production módban van a Billingo akkor a sandboxot még azon IDk alapján sem éred el ami komoly probléma, lévén ha egyszer beüzemelted, onnan kezdve elég körülményesen lehet csak fejleszteni, átalakítani, hiszen le kell állítania  fizetést arra az időre, nehogy valaki csak sandboxról kapjon nem hiteles számlát ameddig valaki éppen fejleszt/tesztel.

Igen, jelenleg a teljes válaszon megyek végig, és kénytelen vagyok a comment mező tartalmát felparseolni hogy megkapjam azt, hogy melyik proformából készült.

**Comment by albertborsos** [comment]
Regelj be egy dev fiókot, ami mindig sandbox módban van :D 

Amúgy a teljes válasz max 50 eredményt tartalmazhat :/ Igaz, meg lehet oldani így is, csak nem túl elegáns :D

Nem lettem boldogabb. Köszi az infókat!

**Comment by albertborsos** [comment]
Most az élesen fejlesztek, ott sem működik. Sőt, már egy standard emailcímre is hibát dob az ügyfél létrehozásnál :D

Inkább rápróbálok a számlázz.hu-ra, minthogy ebbe öljem az órákat...

**Comment by Razoralpha** [comment]
Igen, a dev fiók lett nálunk is a megoldás, az 50 találat nem sok, de a page-el azért végig lehetne léptetni, ha figyelembe venné az említett paramétereket vagy azok nem NULL-ok lennének.

**Comment by albertborsos** [comment]
Nekem az `invoices/query` végpont konstans `null`-t ad vissza, akárhogy hívom meg. Ez a dev fiók úgy tűnik iparági sztenderd, mert a számlázz.hu ezt az opciót adja meg, ha az "API"-ját szeretnéd használni. Azt gyorsan elengedtem. Bízom benne, hogy lesz valami alternatíva július 1-ig...

**Comment by ttanai** [comment]
én így hívom, nekem működik a végpont: 
/api/invoices/query?year_start=2018&year_end=2018

**Comment by Razoralpha** [comment]
Most éppen megy, de nem csak a fenti formában, viszont ez tartott 7-8 hónapig, a supportjuk pedig csak mint feature request volt hajlandó felvenni, és már az is hónapokkal ezelőtt volt. Szerintem aki teheti válasszon mást, nem éri meg azt a mennyiségű hajszálat amit már kitépkedtem miattuk.
